src/OrchardCore.Modules/OrchardCore.ContentFields/Services/DefaultContentPickerResultProvider.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Records;
+using YesSql;
+using YesSql.Services;
+
+namespace OrchardCore.ContentFields.Services
+{
+    public class DefaultContentPickerResultProvider : IContentPickerResultProvider
+    {
+        private readonly IContentManager _contentManager;
+        private readonly ISession _session;
+
+        public DefaultContentPickerResultProvider(IContentManager contentManager, ISession session)
+        {
+            _contentManager = contentManager;
+            _session = session;
+        }
+
+        public string Name => "Default";
+
+        public async Task<IEnumerable<ContentPickerResult>> Search(ContentPickerSearchContext searchContext)
+        {
+            var contentItems = await _session.Query<ContentItem, ContentItemIndex>()
+                .With<ContentItemIndex>(x => x.ContentType.IsIn(searchContext.ContentTypes) && x.Published)
```

Comment:
We need to be able to select draft items, even if the end result will actually load it only if it's published. A referenced item might not yet be visible at the time you are editing this one.

Reply:
I changed this to select latest